### DISCUSSION TITLE
cutlass_profiler build with WMMA API fails?

### DISCUSSION THREAD

**Discussion by navdeepkk** [NVIDIA/cutlass#288] [Q&A]
I was trying to build the culass_profiler to see how GEMM kernels using the WMMA API perform. I removed the comment on this [line](https://github.com/NVIDIA/cutlass/blob/6c29fe20ba9d19c1de0cbe3d409a883f37a1a1a7/tools/library/scripts/generator.py#L1225). Somehow the build fails. Any Idea on what is going on? The issue can be reproduced by simply removing the comment on the line mentioned and then build CUTLASS.

**Comment by hwu36** [comment]
wmma has not benn maintained for a while.  We are welcome the community to fix it.  

If you just want to measure the performance quickly, you can use nvprof/nsight to measure the kernel time of wmma unit tests (https://github.com/NVIDIA/cutlass/tree/master/test/unit/gemm/device).  Or you can modify the wmma unit test to use cudaEvent to measure the runtime of multiple iterations to get more accurate number.

**Comment by tjingrant** [comment]
Is this fixed already? I cannot seem to reproduce the issue. Things I've done:
- Uncomment this [line](https://github.com/NVIDIA/cutlass/blob/6c29fe20ba9d19c1de0cbe3d409a883f37a1a1a7/tools/library/scripts/generator.py#L660). 
- Rebuild CUTLASS from scratch.
- Run profiler with `./tools/profiler/cutlass_profiler --kernels=cutlass_wmma_* --m=4096 --n=4096 --k=4096`.

Everything works, except for a few incorrect results near the end, is that the issue? Or perhaps it only happens with SM75, and I've only have SM70 machines (Volta)?

**Reply by hwu36** [reply]
I don't think we have done anything for it.

**Reply by tjingrant** [reply]
@navdeepkk could you provide more information? What are the failure symptoms? It will help me determine whether I have the right hardware to work on this issue.
